src/Raven.Server/Documents/Expiration/ExpiredDocumentsCleaner.cs

```diff
@@ -90,9 +91,17 @@ internal async Task CleanupExpiredDocs()
 
                 using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
                 {
+                    DatabaseTopology topology;
+                    using (_database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
+                    {
+                        topology = _database.ServerStore.Cluster.ReadDatabaseTopology(serverContext, _database.Name);
+                    }
+
+                    var isFirstInTopology = string.Equals(topology.AllNodes.FirstOrDefault(), _database.ServerStore.NodeTag, StringComparison.OrdinalIgnoreCase);
```

Comment:
I'm wondering if this should not run e.g. on last node. We are performing a lot of operations on first already, e.g. client API is connecting to first by default. 

Also I'm wondering if we should not check only Members?

Reply:
I'm using the first one intentionally, that is the one that we are likely to already be writing for, so the chance of creating a conflict with user writes is even lower.
And we need to take into account that we don't _have_ members, that all have a rehab, so we'll use that.